pika/channel.py

```diff
@@ -753,127 +756,132 @@ def open(self):
         self._add_callbacks()
         self._rpc(spec.Channel.Open(), self._on_openok, [spec.Channel.OpenOk])
 
-    def queue_bind(self, callback, queue, exchange,
+    def queue_bind(self,
+                   queue,
+                   exchange,
                    routing_key=None,
-                   nowait=False,
-                   arguments=None):
+                   arguments=None,
+                   callback=None):
         """Bind the queue to the specified exchange
 
-        :param callable callback: The callback to call on Queue.BindOk;
-            MUST be None when nowait=True.
         :param queue: The queue to bind to the exchange
         :type queue: str or unicode
         :param exchange: The source exchange to bind to
         :type exchange: str or unicode
         :param routing_key: The routing key to bind on
         :type routing_key: str or unicode
-        :param bool nowait: Do not wait for a Queue.BindOk
         :param dict arguments: Custom key/value pair arguments for the binding
+        :param callable callback: The callback to call on Queue.BindOk
+        :raises ValueError:
 
         """
-        self._validate_channel_and_callback(callback)
-        replies = [spec.Queue.BindOk] if nowait is False else []
+        self._validate_channel()
+        nowait = self._validate_rpc_completion_callback(callback)
         if routing_key is None:
             routing_key = queue
         return self._rpc(spec.Queue.Bind(0, queue, exchange, routing_key,
                                          nowait, arguments or dict()),
-                         callback, replies)
+                         callback,
+                         [spec.Queue.BindOk] if not nowait else [])
 
-    def queue_declare(self, callback,
+    def queue_declare(self,
                       queue='',
```

Comment:
Just curious -- how common of usage is it to have the queue auto-named by RabbitMQ?

Reply:
I don't know how common an auto-named queue is, but it is supported and I suspect is why queue defaults to the empty string in the code. I personally think that may be more confusing to a new user than having them pass the empty string explicitly if they want an auto-named queue. I'll see what the `bunny` library does.

Reply:
`bunny` does not use the empty string as the default, which I like better. @vitaly-krugl what do you think?